banyan/resource_service.go

```diff
@@ -1659,3 +1605,266 @@ func resourceServiceDelete(ctx context.Context, d *schema.ResourceData, m interf
+	flattened = append(flattened, v)
+	return
+}
+
+func flattenServicePatterns(toFlatten []service.Pattern) (flattened []interface{}) {
+	for _, item := range toFlatten {
+		v := make(map[string]interface{})
+		v["hosts"] = flattenServiceHosts(item.Hosts)
+		v["mandatory_headers"] = item.MandatoryHeaders
+		v["methods"] = item.Methods
+		v["paths"] = item.Paths
+		v["source_cidrs"] = item.SourceCIDRs
+		flattened = append(flattened, v)
+	}
+	return
+}
+
+func flattenServiceHosts(toFlatten []service.Host) (flattened []interface{}) {
+	for _, item := range toFlatten {
+		v := make(map[string]interface{})
+		v["origin_header"] = item.OriginHeader
+		v["target"] = item.Target
+		flattened = append(flattened, v)
+	}
+	return
+}
+
+func flattenServiceHTTPHealthCheck(toFlatten service.HTTPHealthCheck) (flattened []interface{}) {
+	v := make(map[string]interface{})
+	v["addresses"] = toFlatten.Addresses
+	v["enabled"] = toFlatten.Enabled
+	v["method"] = toFlatten.Method
+	v["from_address"] = toFlatten.FromAddress
+	v["https"] = toFlatten.HTTPS
+	v["path"] = toFlatten.Path
+	v["user_agent"] = toFlatten.UserAgent
+	flattened = append(flattened, v)
+	return
+}
+
+func flattenServiceHTTPRedirect(toFlatten service.HTTPRedirect) (flattened []interface{}) {
+	v := make(map[string]interface{})
+	v["addresses"] = toFlatten.Addresses
+	v["enabled"] = toFlatten.Enabled
+	v["from_address"] = toFlatten.FromAddress
+	v["status_code"] = toFlatten.StatusCode
+	v["url"] = toFlatten.URL
+	flattened = append(flattened, v)
+	return
+}
+
+func flattenServiceOIDCSettings(toFlatten service.OIDCSettings) (flattened []interface{}) {
+	v := make(map[string]interface{})
+	v["api_path"] = toFlatten.APIPath
+	v["enabled"] = toFlatten.Enabled
+	v["post_auth_redirect_path"] = toFlatten.PostAuthRedirectPath
+	v["service_domain_name"] = toFlatten.ServiceDomainName
+	v["suppress_device_trust_verification"] = toFlatten.SuppressDeviceTrustVerification
+	v["trust_callbacks"] = toFlatten.TrustCallBacks
+	flattened = append(flattened, v)
+	return
+}
+
+func flattenServiceClientCIDRs(toFlatten []service.ClientCIDRs) (flattened []interface{}) {
+	for _, item := range toFlatten {
+		v := make(map[string]interface{})
+		v["address"] = flattenServiceCIDRAddresses(item.Addresses)
+		v["clusters"] = item.Clusters
+		v["host_tag_selector"] = item.HostTagSelector
+		flattened = append(flattened, v)
+	}
+	return
+}
+
+func flattenServiceCIDRAddresses(toFlatten []service.CIDRAddress) (flattened []interface{}) {
+	for _, item := range toFlatten {
+		v := make(map[string]interface{})
+		v["cidr"] = item.CIDR
+		v["ports"] = item.Ports
+		flattened = append(flattened, v)
+	}
+	return
+}
+
+func validatePort() func(val interface{}, key string) (warns []string, errs []error) {
+	return func(val interface{}, key string) (warns []string, errs []error) {
+		v, err := typeSwitchPort(val)
+		if err != nil {
+			errs = append(errs, err)
+			return
+		}
+		if v < 0 || v > math.MaxUint16 {
```

Comment:
Should this be `if v <= 0 || v >= math.MaxUint16`? Corresponds to allowed range `[1..65535]`.